tests/unit/Subscription/MembershipTest.php

```diff
@@ -5,14 +5,63 @@
 namespace GroupLife\Core\tests\Subscription;
 
 use GroupLife\Core\Subscription\Membership;
+use GroupLife\Core\Visitor;
 use PHPUnit\Framework\TestCase;
 
 class MembershipTest extends TestCase
 {
+    /**
+     * @throws \Exception
+     */
     public function testScheduleMembershipGetFunctions()
     {
-        $purchase = new Membership(new \DateTimeImmutable('2021-01-01'), new \DateInterval('P1M'));
-        $this->assertEquals(new \DateTimeImmutable('2021-01-01'), $purchase->getStartDay());
-        $this->assertEquals(new \DateInterval('P1M'), $purchase->getPeriod());
+        $this->assertEquals(
+            new \DateTimeImmutable('2021-01-01', new \DateTimeZone('Europe/Berlin')),
+            self::membershipSubscription()->getStartDay()
+        );
+        $this->assertEquals(new \DateInterval('P1M'), self::membershipSubscription()->getPeriod());
+    }
+
+    /**
+     * @throws \Exception
+     */
+    public function testJsonSerialize()
+    {
+        self::assertJsonStringEqualsJsonString(
+            <<<'JSON'
+                {
+                    "id": null,
+                    "type": "GroupLife\\Core\\Subscription\\Membership",
+                    "startDay": {
+                        "date": "2021-01-01 00:00:00.000000",
+                        "timezone_type": 3,
+                        "timezone": "Europe\/Berlin"
+                    },
+                    "period": 2678400,
+                    "activity": null,
+                    "visitor": {
+                        "id": null,
```

Comment:
Building the API we'll describe the data structures in some way. I suspect we'll define "id is an integer", "activity is an object of certain structure". Value `null` won't fit in those types. Let's follow the rule: when we don't know something we don't define it. Just omit the property.

Reply:
I've removed 'activity' form all subscriptions except Activity / Subscription. But about visitor / id: it's still empty because it wasn't saved. I thought API manipulates only saved data